### PR TITLE
Refactor "Naming" of Sidechain Compressor to Sidechain

### PR DESCRIPTION
**Pull request by seangoodvibes**
Refactored references to the word Compressor when used in the Sidechain context to just use the word Sidechain. This will help avoid confusion with the Master Compressor.

This also includes renaming the Sidechain Compressor menu to just say Sidechain

**Only thing not covered by this PR:** there are still references to compressor in the file name strings. I didn't touch them because I didn't want to break file compatibility.

### CODE REVIEW

File: src/deluge/gui/menu_item/patch_cable_strength.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ class PatchCableStrength : public Decimal, public MenuItemWithCCLearning {
 	[[nodiscard]] int32_t getMaxValue() const final { return kMaxMenuPatchCableValue; }
 	[[nodiscard]] int32_t getNumDecimalPlaces() const final { return 2; }
 	virtual int32_t getDefaultEditPos() { return 2; }
-	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange) override;
+	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange);
 	virtual ParamDescriptor getDestinationDescriptor() = 0;
```
Comment (seangoodvibes) [inline src/deluge/gui/menu_item/patch_cable_strength.h:35]: not sure why but it wouldn't compile unless I removed override from this function...

File: src/deluge/gui/menu_item/patch_cable_strength/fixed.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Fixed : public Regular {
 	Fixed(l10n::String newName, int32_t newP = 0, PatchSource newS = PatchSource::LFO_GLOBAL)
 	    : Regular(newName), p(newP), s(newS) {}
 
-	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange) final;
+	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange);
```
Comment (seangoodvibes) [inline src/deluge/gui/menu_item/patch_cable_strength/fixed.h:28]: same thing here, had to remove final to get it to compile...

File: src/deluge/gui/menu_item/patch_cable_strength/regular.h [hunk 3]
```diff
@@ -27,7 +27,7 @@ class Regular : public PatchCableStrength {
 	ParamDescriptor getDestinationDescriptor() final;
 	PatchSource getS() final;
 	ParamDescriptor getLearningThing() final;
-	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange) override;
+	MenuPermission checkPermissionToBeginSession(Sound* sound, int32_t whichThing, MultiRange** currentRange);
```
Comment (seangoodvibes) [inline src/deluge/gui/menu_item/patch_cable_strength/regular.h:30]: and had to remove override here too for same reason

**Review by m-m-adams — COMMENTED** [review]
You need to add the overrides back in and fix the underlying problem. There's two different classes named MultiRange (one in menu item, one in storage) and the change in include order in menus.cpp changes which one gets used. I believe the one we want in checkPermissionToBeginSession is the storage version but I'm not actually sure, so you'll need to investigate that

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**